File: 백준/Gold/1106. 호텔/호텔.cc
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

const int INF = 1e9;

int main() {
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int c, n; cin >> c >> n;
    int dp[1101]; fill(dp, dp + 1101, INF);
    dp[0] = 0;
    while(n--){
        int cst, peo; cin >> cst >> peo;
        for(int i = peo; i <= 1100; i++){
            if(dp[i-peo] != INF) dp[i] = min(dp[i], dp[i-peo]+cst);
        }
    }
    int ans = INF;
    for(int i = c; i <= 1100; i++) if(dp[i] < ans) ans = dp[i];
    cout << ans;
    return 0;
}

File: 백준/Gold/1106. 호텔/호텔.py
import sys
input = lambda: sys.stdin.readline().rstrip()
print = lambda *args, sep=" ", end="\n": sys.stdout.write(sep.join(map(str, args)) + end)

from math import inf

def main():
    c, n = map(int, input().split())
    nc = c+100
    dp = [inf]*(nc+1)
    dp[0] = 0
    for _ in range(n):
        cst, peo = map(int, input().split())
        for i in range(peo, nc+1):
            if dp[i-peo] != inf:
                dp[i] = min(dp[i], dp[i-peo]+cst)
    print(min(dp[c:]))
    return
if __name__ == "__main__":
    main()